test/LightDFGSummaryTypes.jl

```diff
@@ -112,58 +117,60 @@ end
     # @test data(f1) == f1.data
     # @test getData(f1) == f1.data
     # Internal function
-    @test internalId(f1) == f1._internalId
+    if FACTYPE == DFGFactorSummary
+        @test internalId(f1) == f1._internalId
+    end
 
-    @test getSolverParams(dfg) != nothing
-    @test setSolverParams(dfg, getSolverParams(dfg)) == getSolverParams(dfg)
 end
 
 @testset "Updating Nodes" begin
-    global dfg
-    #get the variable
-    var = getVariable(dfg, :a)
-    #make a copy and simulate external changes
-    newvar = deepcopy(var)
-    estimates(newvar)[:default] = Dict{Symbol, VariableEstimate}(
-        :max => VariableEstimate(:default, :max, [100.0]),
-        :mean => VariableEstimate(:default, :mean, [50.0]),
-        :modefit => VariableEstimate(:default, :modefit, [75.0]))
-    #update
-    mergeUpdateVariableSolverData!(dfg, newvar)
-    #TODO maybe implement ==; @test newvar==var
-    Base.:(==)(varest1::VariableEstimate, varest2::VariableEstimate) = begin
-        varest1.lastUpdatedTimestamp == varest2.lastUpdatedTimestamp || return false
-        varest1.ppeType == varest2.ppeType || return false
-        varest1.solverKey == varest2.solverKey || return false
-        varest1.estimate == varest2.estimate || return false
-        return true
+    if VARTYPE == DFGVariableSummary
+        global dfg
+        #get the variable
+        var = getVariable(dfg, :a)
+        #make a copy and simulate external changes
+        newvar = deepcopy(var)
+        estimates(newvar)[:default] = Dict{Symbol, VariableEstimate}(
+            :max => VariableEstimate(:default, :max, [100.0]),
+            :mean => VariableEstimate(:default, :mean, [50.0]),
+            :modefit => VariableEstimate(:default, :modefit, [75.0]))
+        #update
+        mergeUpdateVariableSolverData!(dfg, newvar)
+        # #TODO maybe implement ==; @test newvar==var
+        # Base.:(==)(varest1::VariableEstimate, varest2::VariableEstimate) = begin
+        #     varest1.lastUpdatedTimestamp == varest2.lastUpdatedTimestamp || return false
+        #     varest1.ppeType == varest2.ppeType || return false
+        #     varest1.solverKey == varest2.solverKey || return false
+        #     varest1.estimate == varest2.estimate || return false
+        #     return true
+        # end
+        #For now spot check
+        # @test solverDataDict(newvar) == solverDataDict(var)
+        @test estimates(newvar) == estimates(var)
+
+        # Delete :default and replace to see if new ones can be added
+        delete!(estimates(newvar), :default)
+        estimates(newvar)[:second] = Dict{Symbol, VariableEstimate}(
+            :max => VariableEstimate(:default, :max, [10.0]),
+            :mean => VariableEstimate(:default, :mean, [5.0]),
+            :ppe => VariableEstimate(:default, :ppe, [7.0]))
+
+        # Persist to the original variable.
+        mergeUpdateVariableSolverData!(dfg, newvar)
```

Comment:
just check if this has a relation to discussion at #182 ?

Reply:
Jip.  We had a different view on #166 how it worked. I used merge in LightDFG and GraphDFG, and sam replaced. So he changed it to mergeupdate. I think we can sort it out in #182